src/GraphQl/Action/EntrypointAction.php

```diff
@@ -117,4 +113,48 @@ private function parseRequest(Request $request): array
 
         return [$query, $operation, $variables];
     }
+
+    private function parseInput($query, $variables, $operation, string $jsonContent): array
+    {
+        $input = json_decode($jsonContent, true);
+        if (isset($input['query'])) {
+            $query = $input['query'];
+        }
+
+        if (isset($input['variables'])) {
+            $variables = \is_array($input['variables']) ? $input['variables'] : json_decode($input['variables'], true);
+        }
+
+        if (isset($input['operation'])) {
+            $operation = $input['operation'];
+        }
+
+        return [$query, $operation, $variables];
+    }
+
+    private function applyMappingToVariables(Request $request, array $variables): array
+    {
+        $mapValues = json_decode($request->request->get('map'), true);
+        if (!$mapValues) {
+            return $variables;
+        }
+        foreach ($mapValues as $key => $value) {
+            if ($request->files->has($key)) {
+                foreach ($mapValues[$key] as $mapValue) {
+                    $path = explode('.', $mapValue);
+                    if ('variables' === $path[0]) {
+                        unset($path[0]);
+                        $temp = &$variables;
```

Comment:
Could you try to avoid using references?

Reply:
I can do that but the logic will be complex and it won't be efficient.I'll try to change it.